ezb_api/ezb_api.py

```diff
@@ -0,0 +1,342 @@
+        root = ElementTree.fromstring(response.content)
+        data = []
+        cols = ['TIME_PERIOD', 'OBS_VALUE']
+
+        for obs_element in root.findall('.//ezb:Obs', _XML_NS):
+            obs_dimension = obs_element[0].attrib['value']
+            obs_value = obs_element[1].attrib['value']
+            obs_value_float = convert_to_float_value(obs_value)
+            data.append([obs_dimension, obs_value_float])
+        if not data:
+            raise ValueError(f"No values can be retrieved from API")
+        else:
+            ezb_data = pd.DataFrame(data=data, columns=cols)
+            return ezb_data
+    else:
+        raise ValueError("In response content no XML parseable content contained.")
+
+
+def validate_curr_requirements(source: str, target: str) -> bool:
+    """Validate that currency requirements are satisfied.
+
+    Args:
+        source (str): source currency
+        target (str): target currency
+
+    Returns:
+        A boolean, it is false if currency does not statisfy the requirements and true if it matches the requirements.
+    """
+    validation_result = False
+    if validate_curr(source) and validate_curr(target):
+        if target == _REQUIRED_CURRENCY:
+            if target != source:
+                return True
+    return validation_result
+
+
+def get_exchange_rate(source: str, target: str = "EUR") -> pd.DataFrame:
+    """Get the exchange rate between a currency pair
+
+    Sends API request to EZP, for requesting past exchange rate of a currency pair
+    and returns pandas dataframe with the time period and exchange rate.
+
+    Args:
+        source (str): source currency
+        target (str): target currency, Default to EUR
+
+    Returns:
+        pandas Dataframe with TIME_PERIOD and OBS_VALUE column
+
+    Raises:
+        ValueError: An error occurred if EZB API responds with other status code than 200
+        ValueError: An error occurred if other currencies, string values, int values, bool values are parsed to the method
+    """
+
+    if validate_curr_requirements(source, target):
+        currency_pair_exchange_rate_url = _EXCHANGE_RATE_URL.format(source, target)
+        ezb_request = get_response_from_ezb(currency_pair_exchange_rate_url)
+        if check_request_response(ezb_request):
+            curr_exchange_rate_pdf = get_pdf_from_ezb_xml(ezb_request)
+            return curr_exchange_rate_pdf
+        else:
+            raise ValueError(
+                f"REST API retrieves for {currency_pair_exchange_rate_url} following code: {ezb_request.status_code}")
+
+    else:
+        raise ValueError("Currency does not match the requirements.")
+
+
+def check_identifier(identifier: str):
+    """Check validity of the identifier.
+
+    Args:
+        identifier (str): the identifier given as string
+
+    Returns:
+        A boolean, it is false if identifier does not statisfy the requirements and true if it matches the requirements.
+    """
+    validation_result = False

```

Comment:
Can we please just return False at the end? What's the point of having a variable if you just gonna return it?